rules/xcodeproj.bzl

```diff
@@ -40,9 +42,13 @@ def _xcodeproj_aspect_impl(target, ctx):
             test_env_vars = tuple(env_key_value_pairs.items())
             commandlines_args = getattr(ctx.rule.attr, "args", [])
             test_commandline_args = tuple(commandlines_args)
+            test_host_target = getattr(ctx.rule.attr, "test_host", None)
+            if test_host_target and test_host_target.label.package != "rules/test_host_app":
```

Comment:
nit: should this also check the repository?
(I'm guessing this is here so we don't add in the default test host, or something like that? a comment to that effect would be helpful)

Reply:
I realized that this logic is probably more correct. WDYT?

```
            # The ios_unit_test rule sets the test_host attribute to the rules_ios repo's rules/test_host_app target by default, if test_host = True.
            # The generated xcodeproject should only reference test host applications that live in the same workspace as the test target.
            if test_host_target.label.workspace_root == target.label.workspace_root:
                test_host_appname = test_host_target.label.name
```

Reply:
Will app-hosted tests that have cross-repository deps continue to work if you do that?